pmm/summary.go

```diff
@@ -0,0 +1,270 @@
+		fmt.Println("Error exec pmm-admin list", err)
+		return nil
+	}
+	if strings.Contains(string(cmdPmmList), "mysql:metrics") {
+		monitoredDBServices = append(monitoredDBServices, "mysql")
+	}
+	if strings.Contains(string(cmdPmmList), "mongodb:metrics") {
+		monitoredDBServices = append(monitoredDBServices, "mongodb")
+	}
+
+	return monitoredDBServices
+}
+
+// Copy a file for collecting into the final archive
+func copyFile(dirname, name string) error {
+	srcf, err := os.Open(name)
+	if err != nil {
+		return fmt.Errorf("copy %s to %s: %v", name, dirname, err)
+	}
+	defer srcf.Close()
+
+	dstf, err := os.OpenFile(filepath.Join(dirname, filepath.Base(name)), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0666)
+	if err != nil {
+		return fmt.Errorf("copy %s to %s: %v", name, dirname, err)
+	}
+
+	if _, err := io.Copy(dstf, srcf); err != nil {
+		dstf.Close()
+		defer dstf.Close()
+		os.Remove(dstf.Name())
+		return fmt.Errorf("copy %s to %s: %v", name, dirname, err)
+	}
+
+	if err := dstf.Close(); err != nil {
+		os.Remove(dstf.Name())
+		return fmt.Errorf("copy %s to %s: %v", name, dirname, err)
+	}
+
+	return nil
+}
+
+// zipIt archives collected information.
+func zipIt(source, target string) error {
+	zipfile, err := os.Create(target)
+	if err != nil {
+		return err
+	}
+	defer zipfile.Close()
+	archive := zip.NewWriter(zipfile)
+	defer archive.Close()
+	info, err := os.Stat(source)
+	if err != nil {
+		return err
+	}
+	var baseDir string
+	if info.IsDir() {
+		baseDir = filepath.Base(source)
+	}
+	err = filepath.Walk(source, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		header, err := zip.FileInfoHeader(info)
+		if err != nil {
+			return err
+		}
+		if baseDir != "" {
+			header.Name = filepath.Join(baseDir, strings.TrimPrefix(path, source))
+		}
+		if info.IsDir() {
+			header.Name += "/"
+		} else {
+			header.Method = zip.Deflate
+		}
+		writer, err := archive.CreateHeader(header)
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
+		}
+		file, err := os.Open(path)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+		_, err = io.Copy(writer, file)
+		return err
+	})
+	return err
+}
+
+// CollectSummary get output of system and pmm utilites.
+func (a *Admin) CollectSummary() error {
+	fmt.Println("\nCollecting information for system diagnostic")
+	// Create a directory for collecting files and log file for possible errors
+	currentTime := time.Now().Local()
+	cmdHostname, err := os.Hostname()
+	if err != nil {
+		fmt.Println("Error getting hostname:", err)
+		cmdHostname = ("unknown")
+	}
+
+	dirname := strings.Join([]string{"/tmp/pmm", cmdHostname, currentTime.Format("2006-01-02T15_04_05")}, "-")
+	err = os.MkdirAll(dirname, 0755)
+	if err != nil {
+		fmt.Printf("Error creating a temporary directory %s: %v\n", dirname, err)
+		os.Exit(1)
+	}
+
+	dstLogInfo, _ := os.OpenFile(filepath.Join(dirname, "pmm-summary.err"), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0666)
```

Comment:
os.Create().
Also, do not ignore the error. If the file can't be created, the program will crash on the next line.